dcos/http.py

```diff
@@ -120,6 +133,15 @@ def _request(method,
     :rtype: Response
     """
 
+    if timeout is True:
+        if toml_config is None:
+            toml_config = config.get_config()
+
+        timeout = config.get_config_val("core.timeout", toml_config)
+        timeout = (DEFAULT_CONNECT_TIMEOUT, timeout or DEFAULT_TIMEOUT)
```

Comment:
If `timeout` is `0` or `None`, I believe that this will set `timeout` to `(DEFAULT_CONNECT_TIMEOUT, (DEFAULT_CONNECT_TIMEOUT, DEFAULT_READ_TIMEOUT))`.

That is to say, I think that instead of `DEFAULT_TIMEOUT`, this should say `DEFAULT_READ_TIMEOUT`.

This indicates that there is a missing test.

It is not clear from the public documentation that giving a value of `0` uses the default timeout.

Reply:
hum.. I'll fix this and add some tests :+1: